vaas/vaas/vcl/templates/vcl_blocks/4.0/FLEXIBLE_ROUTER.tvcl

```diff
@@ -20,7 +20,7 @@
 {% for domain, redirect_list in redirects.items() %}
     if (req.http.host == "{{ domain }}") {
 {% for redirect in redirect_list %}
-    {% if loop.index > 1 %}    else if {% else %}    if {% endif %}({{ redirect.condition }}) {
+    {% if loop.index > 1 %}    else if {% else %}    if {% endif %}({{ redirect.condition }}{% if redirect.custom_header_condition %} && req.http.{{ redirect.custom_header }}{% endif %}) {
```

Comment:
I suggest composing the final condition in a dedicated method  of the Redirect class, for example:
```
@property
def final_condition(self):
    if self.required_custom_header:
       return f'{self.condition} && req.http.{self.custom_header}'
    return self.condition
```